metaflow/plugins/aws/step_functions/step_functions_client.py

```diff
@@ -0,0 +1,80 @@
+from metaflow.metaflow_config import get_authenticated_boto3_client
```

Comment:
Higher level file comment -
Could we rename this file to be step_functions.py instead? If Meson would ever be OSS one day; it'd be nice to not have confusion that `meson_client` is actually a way to call Meson within user_code; and I'm not sure this file is exactly in the same spirit.
We can chat more offline if we could split the existing `step_functions.py` that you have into something like `step_functions_api.py` since it has definitions of some objects like `Workflow` etc.

Reply:
Discussed offline - seems the current (code) structure is in line with AWS Batch code. So resolving this one.

Reply:
Meson and AWS Step Functions are two entirely different systems with entirely different integration code, so it doesn't make much sense to have the same filenames. Also, top-level imports from metaflow are not dependent on actual file names. We do `from metaflow import Batch` and not `from metaflow.plugins.aws.batch import Batch`.